tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -110,466 +62,320 @@ var _ = Describe("odo devfile push command tests", func() {
 			helper.Cmd("odo", "push", "--project", commonVar.Project).ShouldPass()
 		})
 
-		It("checks that odo push works with a devfile with sourcemapping set", func() {
-			helper.Cmd("odo", "create", "nodejs", "--project", commonVar.Project, cmpName).ShouldPass()
+		When("JSON output is requested", func() {
+			It("should display output in JSON format", func() {
 
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfileSourceMapping.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
+				helper.Cmd("odo", "create", "nodejs", "--project", commonVar.Project, cmpName).ShouldPass()
```

Comment:
This line is appearing in one branch above it (line no. 38). Overall, this line is appearing in the file for 22 times. I think we can have more `BeforeEach` to handle that.

Reply:
https://github.com/openshift/odo/pull/4820#issuecomment-866813087